code/iomethods.py

```diff
@@ -4,16 +4,39 @@
 import string
 import subprocess
 import sys
+import math
 from datetime import timedelta, datetime
 from os import path
 
 import numpy as np
 import numpy.ma as ma
-from netCDF4 import Dataset
+import netCDF4
 
 import utils
 import variables
 
+def get_fileList_for_binaries(dirPath, startTime, endTime):
+    '''
+
+    '''
+
+    fileList = glob.glob(dirPath+'/*-pixel')
+    fileList.sort()
+
+    newFileList = []
+
+    for file in fileList:
+        dateFromFileName = [token for token in file.split('_') if token.isdigit()]  # Parse date from MERG binary file name
+
+        dateAsDateTime = datetime.strptime(dateFromFileName[0], '%Y%m%d%H')
+        start = datetime.strptime(startTime, '%Y%m%d%H%S')  # Convert start and end times to datetimes to compare with
```

Comment:
@caocampb The variables start and end need to be moved up out of this loop.  You need to compute them once initially and use them for each loop through the fileList.  The code right now will recompute these values (which shouldn't change) each time a new file is parsed.  This seems like a trivial issue, but it's really important to ensure you aren't doing any extra computation within a for loop.